addon/services/router-scroll.js

```diff
@@ -15,6 +15,7 @@ export default Service.extend({
   targetElement: null,
   delayScrollTop: false,
   isFirstLoad: true,
+  preserveScrollPosition: false,
```

Comment:
I wonder if we should reset this somewhere in the `update` method.  What do you think?

Reply:
hey @snewcomer - sorry for the delay, but I'm now back on this :)

I'm not sure what you mean here - are you suggesting that update could reset so the developer wouldn't need to call `routerScroll.set('preserveScrollPosition', false);` after a transition occurs?

I don't think that would work since `update` is called on `_routeWillChange` but we need to know the result of `preserveScrollPosition` in `_routeDidChange` - which means it would have already been reset.

It's possible that `_routeDidChange` could handle resetting after the transition and scroll has occurred - but IMO I think the toggling on/off of `preserveScrollPosition` at the call site makes for more readable and easier to reason about behaviour.

If the call site only mentions disabling, but not then re-enabling `preserveScrollPosition` then all dev's need to be aware about intimate details of the add-on's behaviour to properly read the behaviour of the code in the host app.

Reply:
If the addon sets `preserveScrollPosition` on a singleton, and you are on a route that shouldn't preserve the scroll position, what happens?  Is `preserveScrollPosition` still `true`?

Reply:
Not sure I follow - Do you mean if the controller or QP has `preserveScrollPosition = false` & the service has `preserveScrollPosition = true`?

If so, the value in the service will win - `preserveScrollPosition` will be true.

Reply:
Right.  Ok I think I am talking myself in circles.  You are right.  If the user is going to explicitly set it, they should explicitly unset it.